src/components/autocomplete/js/autocompleteController.js

```diff
@@ -531,7 +531,14 @@ function MdAutocompleteCtrl ($scope, $element, $mdUtil, $mdConstant, $mdTheming,
    * @returns {*}
    */
   function getDisplayValue (item) {
-    return $q.when(getItemText(item) || item);
+    var itemText = getItemText(item) || item;
+
+    if (itemText && !angular.isString(itemText)) {
```

Comment:
I'm not really sure about that check, because `$scope.itemText` can also return a promise as a value.
> http://codepen.io/DevVersion/pen/KrGxaQ (Simple example)

I think it would be better to chain into the returned promise and add the check there.